### ISSUE TITLE
force SFSortingExtractors to have self._sampling_frequency

### ISSUE THREAD

**Issue by alejoe91** [flatironinstitute/spikeforest_old#66]
With this ST update (https://github.com/SpikeInterface/spiketoolkit/pull/177) the comparison functions take a `delta_time` rather than `delta_frames`.

This requires the `SortingExtractor` to have a `self._samapling_frequency` which is returned by the `get_sampling_frequency` functions. All of the updated extractors implement this automatically.

The only issue is the MdaSortingExtractor, because it doesn't save the sampling frequency.

@magland @jamesjun maybe we could add this info to the .mda header?

**Comment by alejoe91** [comment]
something like this https://github.com/SpikeInterface/spikeextractors/pull/196 would fix the MdaSortingExtractor. In general I think that it's always good to have the sampling frequency information for sorting extractors

**Comment by magland** [comment]
I agree it is always good to have sampling freq info in the sorting extractors. However, it is too late to build it into the firings.mda format. Instead we should explicitly pass samplerate=.... into the constructor of the MdaSortingExtractor. The info can be obtained from the accompanying recording extractor.

**Comment by magland** [comment]
Also, I don't like the following in SortingExtractor.py:

```
def get_sampling_frequency(self):
        return self._sampling_frequency
```

The mechanism for a sorting extractor to specify the sampling frequency should not be to set a private variable. Rather it should be to call a set_sampling_frequency(...) --- or to overload the default behavior.

**Comment by alejoe91** [comment]
Ok! I will fix that and apply the changes to spikesorters so that when the output is returned it loads the sampling frequency.

We have a  `set_sampling_frequency`, and that it's the mechanism that the user should use to load sampling frequency info. Internally `set_sampling_frequency` changes the `self._sampling_frequency`

**Comment by magland** [comment]
Okay makes sense. We should just make sure that nothing else sets
._sampling_frequency (always use the set_sampling_frequency method).
